File: maxStack/stack.py
class MaxStack():
    def __init__(self):
        self.items = []

    def push(self, item):
        self.items.append(item)

    def pop(self):
        return self.items.pop(len(self.items) - 1)

    def max(self):
        if(len(self.items) == 0):
            return None
        return max(self.items)

if __name__ == '__main__':
    s = MaxStack()
    print(s.max()) # None

    s.push(1)
    s.push(2)
    s.push(3)
    s.push(2)
    print(s.max()) # 3

    s.pop()
    s.pop()

    print(s.max()) # 2

File: maxStack/stack.js
class MaxStack {
  constructor() {
    this.items = [];
  }

  push(item) {
    this.items.push(item);
  }

  pop() {
    return this.items.splice(this.items.length - 1);
  }

  max() {
    if (this.items.length === 0) {
      return null;
    }

    return this.items.sort()[this.items.length - 1];
  }
}

stack = new MaxStack();
console.log(stack.max()); // Null

stack.push(1);
stack.push(2);
stack.push(3);
stack.push(2);
console.log(stack.max()); // 3

stack.pop();
stack.pop();

console.log(stack.max()); // 2
